lib/webpush_encryption.dart

```diff
@@ -42,106 +43,49 @@ class WebPush {
           'something is wrong'); // TODO, throw more granular errors, subclass them under DecryptionError so applications can just catch DecryptionError
     }
   }
-}
-
-/// Stores the public, private, auth key needed for webpush.
-class WebPushKeys {
-  final List<int> _pubKey;
-  final Uint8List _authKey;
-  final List<int> _privKey;
 
-  //WebPushKeys(this._pubKey, this.authKey, this._privKey); // FUTURE PERSON: don't do positional because messing up the order of the args is very easy
-  WebPushKeys(
-      {required List<int> pubKeyBytes,
-      required Uint8List authKeyBytes,
-      required List<int> privKeyBytes})
-      : _pubKey = pubKeyBytes,
-        _privKey = privKeyBytes,
-        _authKey = authKeyBytes;
-
-  static Future<WebPushKeys> fromMap(Map<String, List<int>> keys) async {
-    var pub = keys['p256dh'], priv = keys['priv'], auth = keys['auth'];
+  /// Encrypts [plaintext] following RFC8291.
+  ///
+  /// Uses public and private [serverKeys] with public and authkey [clientKeys].
+  ///
+  /// [salt] is optional and will be filled with cryptographically random bytes
+  /// if it is not provided. This is mostly useful for debugging.
+  static Future<Uint8List> encrypt({
+    required WebPushKeys serverKeys,
+    required WebPushKeys clientKeys,
+    required Uint8List plaintext,
+    Uint8List? salt,
+  }) async {
+    try {
+      if (salt == null) {
+        salt = Uint8List(16);
+        fillRandomBytes(salt);
+      }
+      final uaPubKey = await clientKeys.publicKey!.pubKey;
+      final asPubKey = await serverKeys.publicKey!.pubKey;
```

Comment:
```suggestion
      final asPubKey = await serverKeys.publicKey.pubKey;
```

As we have the server's key set, there will always be a pubkey and a privkey

Reply:
Todo in other occurrences of course